decimal_test.go

```diff
@@ -32,6 +32,16 @@ var testTable = map[float64]string{
 	.1000000000000008:   "0.1000000000000008",
 }
 
+var testTableScientificNotation = map[string]string{
+	"1e9":       "1000000000",
+	"2.41E-3":   "0.00241",
+	"24.2E-4":   "0.00242",
+	"243E-5":    "0.00243",
+	"-1e-5":     "-0.00001",
```

Comment:
You should not test negatives directly, you should add negative versions of all of these to testTableScientificNotation in the init() function